src/test/regress/sql/pg17.sql

```diff
@@ -0,0 +1,74 @@
+--
+-- PG17
+--
+CREATE SCHEMA pg17;
+SET search_path TO pg17;
+SET citus.next_shard_id TO 917000;
+ALTER SEQUENCE pg_catalog.pg_dist_colocationid_seq RESTART 1400000;
+SET citus.shard_count TO 1;
+SET citus.shard_replication_factor TO 1;
+
+-- PostgreSQL 17 now allows exclusion constraints on partitioned tables, lifting the previous restriction.
+ALTER SEQUENCE pg_catalog.pg_dist_shardid_seq RESTART 5420000;
+ALTER SEQUENCE pg_catalog.pg_dist_placement_placementid_seq RESTART 5420000;
+
+CREATE SCHEMA AT_AddConstNoName;
+
+-- Check "ADD PRIMARY KEY"
+CREATE TABLE AT_AddConstNoName.products (
+    product_no integer,
+    name text,
+    price numeric
+);
+
+SELECT create_distributed_table('AT_AddConstNoName.products', 'product_no');
+
+ALTER TABLE AT_AddConstNoName.products ADD PRIMARY KEY(product_no);
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+	      WHERE rel.relname = 'products';
+
+-- Test with partitioned citus local table
+CREATE TABLE AT_AddConstNoName.citus_local_partitioned_table (dist_col int, another_col int, partition_col timestamp) PARTITION BY RANGE (partition_col);
+CREATE TABLE AT_AddConstNoName.p1 PARTITION OF AT_AddConstNoName.citus_local_partitioned_table FOR VALUES FROM ('2021-01-01') TO ('2022-01-01');
+CREATE TABLE AT_AddConstNoName.longlonglonglonglonglonglonglonglonglonglonglonglonglonglongabc PARTITION OF AT_AddConstNoName.citus_local_partitioned_table FOR VALUES FROM ('2020-01-01') TO ('2021-01-01');
+SELECT citus_add_local_table_to_metadata('AT_AddConstNoName.citus_local_partitioned_table');
+
+\c - - :master_host :master_port
+-- Check "ADD EXCLUDE" errors out for partitioned table since the postgres does not allow it before PG 17
+ALTER TABLE AT_AddConstNoName.citus_local_partitioned_table ADD EXCLUDE(partition_col WITH =);
```

Comment:
Given that this is now something new that Citus supports, we don't necessarily need to add the _exact same_ test that was in `multi_alter_table_add_constraints_without_name.sql`

If we consider this as a new feature in Citus (even though it doesnt need extra code work, apparently), we can create a "simpler" test, e.g. with the following steps:

Create distributed partitioned table dist_blabla
create partitioned citus local table citus_local_blabla
add exclude constraint with name in dist_blabla
check that its propagated to the workers properly
add exclude constraint with name in citus_local_blabla
check that it was added correctly
add exclusin constraints without name in both tables
verify previous command worked
alter the constraints (if meaningful?)
verify previous command worked
drop constraints
verify previous command worked
something else you can think of?

Reply:
@naisila 

I’ve prepared a draft of the test based on this plan. Could you review it, please?
Should this test be in the pg17.sql file?

```
-- Test for exclusion constraints on partitioned and distributed partitioned tables in Citus environment

-- Step 1: Create a distributed partitioned table
CREATE TABLE distributed_partitioned_table (
    id serial PRIMARY KEY,
    partition_col int NOT NULL
) DISTRIBUTED BY (id)
PARTITION BY RANGE (partition_col);

-- Step 2: Create a partitioned Citus local table
CREATE TABLE local_partitioned_table (
    id serial PRIMARY KEY,
    partition_col int NOT NULL
) PARTITION BY RANGE (partition_col);

-- Step 3: Add an exclusion constraint with a name to the distributed partitioned table
ALTER TABLE distributed_partitioned_table ADD CONSTRAINT dist_exclude_named EXCLUDE USING gist (partition_col WITH =);

-- Step 4: Verify propagation of exclusion constraint to worker nodes
\c - - :public_worker_1_host :worker_1_port
SELECT * FROM pg_dist_shard_exclude WHERE constraint_name = dist_exclude_named;

-- Step 5: Add an exclusion constraint with a name to the Citus local partitioned table
\c - - :master_host :master_port
ALTER TABLE local_partitioned_table ADD CONSTRAINT local_exclude_named EXCLUDE USING gist (partition_col WITH =);

-- Step 6: Verify the exclusion constraint on the local partitioned table
SELECT conname FROM pg_constraint WHERE conname = 'local_exclude_named';

-- Step 7: Add exclusion constraints without names to both tables
ALTER TABLE distributed_partitioned_table ADD EXCLUDE USING gist (partition_col WITH =);
ALTER TABLE local_partitioned_table ADD EXCLUDE USING gist (partition_col WITH =);

-- Step 8: Verify the unnamed exclusion constraints were added
SELECT * FROM pg_constraint WHERE conrelid = 'local_partitioned_table'::regclass;
\c - - :public_worker_1_host :worker_1_port
SELECT * FROM pg_constraint WHERE conrelid = 'distributed_partitioned_table'::regclass;

-- Step 9: Drop the exclusion constraints from both tables
\c - - :master_host :master_port
ALTER TABLE distributed_partitioned_table DROP CONSTRAINT dist_exclude_named;
ALTER TABLE local_partitioned_table DROP CONSTRAINT local_exclude_named;

-- Step 10: Verify the constraints were dropped
SELECT * FROM pg_constraint WHERE conname = 'dist_exclude_named';
SELECT * FROM pg_constraint WHERE conname = 'local_exclude_named';

-- Step 11: Clean up - Drop the tables
DROP TABLE IF EXISTS distributed_partitioned_table;
DROP TABLE IF EXISTS local_partitioned_table;

```

Reply:
Very nice tests!

> Should this test be in the pg17.sql file?

Yes

Reply:
tests are added